catroid/src/main/java/org/catrobat/catroid/content/actions/EditLookAction.kt

```diff
@@ -66,32 +66,61 @@ class EditLookAction : PocketPaintAction() {
         }
     }
 
+    override fun restart() {
+        val stageActivity = StageActivity.activeStageActivity.get()
+        if (stageActivity != null && scope?.sprite?.isClone == true) {
+            protectOriginalLookData()
+        }
+        super.restart()
+    }
+
     override fun onIntentResult(resultCode: Int, data: Intent?) {
         val stageActivity = StageActivity.activeStageActivity.get()
         if (resultCode == Activity.RESULT_OK && stageActivity != null) {
             setLookData()
         }
         LookRequester.anyAsked = false
         responseReceived = true
-        stageActivity?.onResume()
+    }
+
+    @VisibleForTesting
+    fun protectOriginalLookData() {
+        val sprite = scope?.sprite ?: return
+        val oldLookData = sprite.look?.lookData ?: return
+        val lookDataName = oldLookData.name ?: return
+        val oldLookDataFile = sprite.look?.lookData?.file ?: return
+
+        try {
+            val newLookDataFile = StorageOperations.duplicateFile(oldLookDataFile)
+            val newLookData = LookData(lookDataName, newLookDataFile)
+            newLookData.isCopy = true
+            sprite.lookList.remove(oldLookData)
+            sprite.lookList.add(newLookData)
+            sprite.look.lookData = newLookData
+        } catch (e: IOException) {
+            Log.e(TAG, Log.getStackTraceString(e))
+        }
     }
 
     @VisibleForTesting
     fun setLookData() {
         val sprite = scope?.sprite ?: return
-        val lookData = sprite?.look?.lookData ?: return
-        val lookDataName = lookData.name ?: return
-        val lookDataOldFile = sprite.look?.lookData?.file ?: return
+        val tempLookData = sprite.look?.lookData ?: return
+        val lookDataName = tempLookData.name ?: return
+        val tempLookDataFile = sprite.look?.lookData?.file ?: return
 
         try {
-            val lookDataNewFile = StorageOperations.duplicateFile(lookDataOldFile)
-            val lookData = LookData(lookDataName, lookDataNewFile)
-            val lookDataIndex = sprite.lookList.indexOf(sprite.look.lookData)
-            sprite.look.lookListIndexBeforeLookRequest = lookDataIndex
-            sprite.lookList.removeAt(lookDataIndex)
-            sprite.lookList.add(lookDataIndex, lookData)
-            lookDataOldFile.delete()
-            lookData.collisionInformation.calculate()
+            val newLookDataFile = StorageOperations.duplicateFile(tempLookDataFile)
+            val newLookData = LookData(lookDataName, newLookDataFile)
+            if (sprite.isClone) {
+                newLookData.isCopy = true
+            } else {
+                tempLookDataFile.delete()
```

Comment:
When you delete the original here and other sprites/clone still use it the project may crash. Please defer the cleanup to when the stage is finished/cleaned up.
[EditLookCloneCrashes.zip](https://github.com/Catrobat/Catroid/files/9927858/EditLookCloneCrashes.zip)